src/main/java/uk/gov/companieshouse/overseasentitiesapi/utils/PublicPrivateDataCombiner.java

```diff
@@ -151,4 +167,61 @@ public Map<String, Pair<CompanyOfficerApi, ManagingOfficerDataApi>> buildMergedM
 
     return combineMO;
   }
+
+  public String logCollatedData() {
+    var result = new StringBuilder("{");
+
+    result.append("\"Managing Officer Data\": [");
+
+    Iterator<Entry<String, Pair<CompanyOfficerApi, ManagingOfficerDataApi>>> moIterator = this.combineMO.entrySet().iterator();
+    while (moIterator.hasNext()) {
+      Entry<String, Pair<CompanyOfficerApi, ManagingOfficerDataApi>> officer = moIterator.next();
+      String[] linkAsArray = officer.getValue().getLeft().getLinks().getSelf().split("/");
+      String publicMoId = linkAsArray[linkAsArray.length - 1];
+
+      result.append("{");
+      result.append("\"Map Key\": \"" + officer.getKey() + "\", ");
+      result.append("\"Public Hashed ID\": \"" + publicMoId + "\", ");
+      result.append("\"Private ID\": \"" + officer.getValue().getRight().getManagingOfficerId() + "\"");
+      result.append("}");
+
+      if (moIterator.hasNext()) {
+        result.append(", ");
+      }
+    }
+    result.append("],");
+
+    result.append("\"Beneficial Owner Data\": [");
+    var boEntries = this.combineBO.entrySet();
+
+    Iterator<Entry<String, Pair<PscApi, PrivateBoDataApi>>> boIterator = boEntries.iterator();
+    while(boIterator.hasNext()) {
```

Comment:
foreach

Reply:
Add the , before you add an item - except for first when none previously added. Instead of adding after if not last. I just think a foreach would be a lot clearer.